$NetBSD: patch-aa,v 1.4 2007/09/22 04:20:01 bsadewitz Exp $

X-Git-Url: http://gitweb.freedesktop.org/?p=pixman.git;a=commitdiff;h=6b96b62ca419ac414d5715a4ba9f30664e510fcd

Don't copy unused bits to an alpha channel

This bug was identified by cairo's rgb24-ignore-alpha test which
blends RGB24 over ARGB32 and notices that "alpha" values are
making it from the source to the destination.
---
From: Jeff Muizelaar <jeff@freiheit.infidigm.net>
Date: Thu, 20 Sep 2007 19:40:17 +0000 (-0400)
Subject: Fix special case selection when the mask has a transform
X-Git-Url: http://gitweb.freedesktop.org/?p=pixman.git;a=commitdiff;h=5b60c91fd6865021aa6027ee65fa8371a4e2d297

Fix special case selection when the mask has a transform

http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=c19ece1d8c32dc81740a4036a642661f54064e75
was a mismerge that avoided setting the maskTransform condition when the mask
has a transform.  This allowed a special case routine to be chosen when the
mask had a transform, which is not expected by the special case routines.
---

--- pixman/pixman-pict.c.orig	2007-07-16 15:05:16.000000000 -0400
+++ pixman/pixman-pict.c
@@ -1426,6 +1426,7 @@ pixman_image_composite (pixman_op_t     
     {
 	maskRepeat = pMask->common.repeat == PIXMAN_REPEAT_NORMAL;
 
+	maskTransform = pMask->common.transform != 0;
 	if (pMask->common.filter == PIXMAN_FILTER_CONVOLUTION)
 	    maskTransform = TRUE;
 
@@ -1778,7 +1779,6 @@ pixman_image_composite (pixman_op_t     
 		    break;
 		case PIXMAN_x8r8g8b8:
 		    switch (pDst->bits.format) {
-		    case PIXMAN_a8r8g8b8:
 		    case PIXMAN_x8r8g8b8:
 #ifdef USE_MMX
 			if (pixman_have_mmx())
@@ -1790,7 +1790,6 @@ pixman_image_composite (pixman_op_t     
 		    }
 		case PIXMAN_x8b8g8r8:
 		    switch (pDst->bits.format) {
-		    case PIXMAN_a8b8g8r8:
 		    case PIXMAN_x8b8g8r8:
 #ifdef USE_MMX
 			if (pixman_have_mmx())
